src/bus_monitor.cpp

```diff
@@ -670,129 +670,67 @@ void SystemStatus::listenBootProgressState()
         });
 }
 
-void SystemStatus::powerPolicyStateCallback(sdbusplus::message::message& msg)
+void SystemStatus::biosAttribsCallback(sdbusplus::message::message& msg)
 {
-    std::string object{};
-    types::ItemInterfaceMap invItemMap;
-
-    msg.read(object, invItemMap);
-    const auto itr = invItemMap.find("PowerRestorePolicy");
-    if (itr != invItemMap.end())
+    if (msg.is_method_error())
     {
-        if (auto powerState = std::get_if<std::string>(&(itr->second)))
-        {
-            // std::cout << "power state = " << *powerState << std::endl;
-            powerPolicy = *powerState;
-
-            setSystemCurrentOperatingMode();
-        }
-        else
-        {
-            std::cerr << "Failed to read power policy from Dbus" << std::endl;
-        }
+        std::cerr << "Error in reading BIOS attribute signal " << std::endl;
+        return;
     }
-}
-
-void SystemStatus::rebootPolicyStateCallback(sdbusplus::message::message& msg)
-{
-    std::string object{};
-    types::ItemInterfaceMap invItemMap;
 
-    msg.read(object, invItemMap);
-    const auto itr = invItemMap.find("AutoReboot");
-    if (itr != invItemMap.end())
+    std::string object;
+    types::BiosBaseTableType propMap;
+    msg.read(object, propMap);
+    for (auto property : propMap)
     {
-        if (auto rebootState = std::get_if<bool>(&(itr->second)))
+        if (property.first == "BaseBIOSTable")
         {
-            // std::cout << "reboot state = " << *rebootState << std::endl;
-            rebootPolicy = *rebootState;
-
-            setSystemCurrentOperatingMode();
-        }
-        else
-        {
-            std::cerr << "Failed to read reboot policy from Dbus" << std::endl;
+            auto boisBaseTable = std::get<0>(property.second);
+            for (const auto& biosItem : boisBaseTable)
+            {
+                auto attributeName = std::get<0>(biosItem);
+                if (attributeName == "pvm_system_operating_mode")
+                {
+                    auto attrValue = std::get<5>(std::get<1>(biosItem));
+                    if (auto val = std::get_if<std::string>(&attrValue))
+                    {
+                        stateManager->setSystemOperatingMode(*val);
+                    }
+                    else
+                    {
+                        std::cerr << "Error reading bios attribute for system "
+                                     "operating mode"
+                                  << std::endl;
+                    }
+                }
+            }
         }
     }
 }
 
-void SystemStatus::listenSystemOperatingModeParameters()
+void SystemStatus::listenSystemOperatingMode()
 {
-    static auto sigPowerPolicy = std::make_unique<sdbusplus::bus::match::match>(
+    static auto biosMatcher = std::make_unique<sdbusplus::bus::match::match>(
```

Comment:
DO we need this local variable "biosMatcher"?

Reply:
It is not just a local variable.